### PR TITLE
COMMERCE-5398 When coupon code is added to an order by a non eligible account an error message should be shown

### PR DESCRIPTION
**Pull request by riccardo-alberti**

### CODE REVIEW

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:stop

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:stop

**Review by dsitu — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountV2Test.java [hunk 1]
```diff
@@ -1469,6 +1471,70 @@ public void testTierPriceEntryAndSystemDiscounts() throws Exception {
...
+			"If a discount rule is not valid the discount is not applied"
+		).given(
+			"A discount with a rule on the cart total"
+		).when(
+			"The price of the product is discovered without and open order"
+		).then(
+			"The discount is not applied"
+		);
+
+		CommerceCatalog catalog =
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountV2Test.java:1486]:
```suggestion
		CommerceCatalog commerceCatalog =
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Review by dsitu — APPROVED** [review]

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:forward

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:forward

**Pull request closed without merging**